plugin/aws/s3Object.go

```diff
@@ -23,14 +22,29 @@ type s3Object struct {
 	client *s3Client.S3
 }
 
-func newS3Object(name string, bucket string, key string, client *s3Client.S3) *s3Object {
+func newS3Object(rawObj *s3Client.Object, name string, bucket string, key string, client *s3Client.S3) *s3Object {
 	o := &s3Object{
 		EntryBase: plugin.NewEntry(name),
 		bucket:    bucket,
 		key:       key,
 		client:    client,
 	}
-	o.DisableCachingFor(plugin.Metadata)
+	o.DisableCachingFor(plugin.MetadataOp)
+
+	attr := plugin.EntryAttributes{}
+	mtime := awsSDK.TimeValue(rawObj.LastModified)
+	attr.SetCtime(mtime)
+	attr.SetMtime(mtime)
+	attr.SetAtime(mtime)
+	// TODO: Export a mungeSize helper to abstract away the common
+	// logic of validating a negative size
+	attr.SetSize(uint64(awsSDK.Int64Value(rawObj.Size)))
+	o.SetInitialAttributes(attr)
+
+	o.Sync(plugin.CtimeAttr(), "LastModified")
```

Comment:
Every plugin has some unenforced tie between Sync name and the result of Metadata. I'd like to come up with a way to simply validate those are consistent, but not seeing anything besides using it against a real (or mocked) API.

Reply:
I mean: something that makes sure that the string you pass in for `Sync` actually matches how you get it from `Metadata`.

Reply:
Ah yeah. In that case, I don't know of another way either